packages/fractal-page-object/src/utils.ts

```diff
@@ -0,0 +1,41 @@
+import { DOM_QUERY, WithElement } from './-private/types';
+
+import type { default as PageObject } from './page-object';
+
+/**
+ * Useful for providing clarity to consumers of page-objects
+ * to provide additional context so "can't access property on undefined"
+ * errors do not public up to the consumer.
+ *
+ * In typescript, this is also useful for type-narrowing so that
+ * you can pass on the element to other utilities.
+ *
+ * @example
+ * ```ts
+ * let page = new Page();
+ *
+ * assertExists('is the element on the page?', page);
+ *
+ * await click(page.element);
+ * ```
+ *
+ * @param {string} msg a descriptor for what it could mean when the element doesn't exist
+ * @param {PageObject} pageObject the page object
+ */
+export function assertExists(
+  msg: string,
+  pageObject: PageObject
+): asserts pageObject is PageObject & WithElement {
+  if (!pageObject.element) {
+    throw new Error(
+      `${msg} >> Tried selector \`${getDescription(pageObject)}\``
+    );
+  }
+}
+
+/**
+ * Utility to get the fully resolved selector path of a {@link PageObject}
+ */
+export function getDescription(pageObject: PageObject): string {
```

Comment:
Can I just express my sincere and emphatic support of `getDescription` I've run into this many times and every time I come back disappointing that there is not a clean way to get the current selector string. The two main cases I've discovered so far are…

In debugging. `Element not found` is not very easy to discover where it is while `selector('… … …')` can really help someone staring at the Browser's console inspector to find where the typo is.

When using 3rd party test helpers. For example, ember-power-calendar has a `calendarSelect()` helper where the first param must be a string because it concatenates its own nested selectors on the one you passed. But as is now we can't convert a fractal-p-o selector back into the full selector from it's `.element` property.

Having this one escape hatch would make composing Page Objects not only easier to debug but also encourage the use of PageObjects like [Martin Fowler](https://www.martinfowler.com/bliki/PageObject.html) originally envisioned by allowing developers to design a fractal-p-o with better methods especially methods that hook into 3rd party test helpers.

```js
class Foo extends PageObject {
  form = selector('[data-test-selector=my-form]', class extends PageObject {
    calendar = selector('ember-power-calendar', class extends PageObject {
      selectDate(date) {
        return calendarSelect(getDescription(this), date);
      }
    });
  });
}
```

Reply:
Related: https://github.com/cibernox/ember-power-calendar/pull/291